app/services/billable_metrics/aggregations/weighted_sum_service.rb

```diff
@@ -26,6 +26,14 @@ def breakdown
       private
 
       def fetch_events(from_datetime:, to_datetime:)
+        if billable_metric.recurring?
+          # NOTE: When recurring we need to scope the fetch using the external ID to handle events
+          #       sent to upgraded/downgraded subscription
+          return recurring_events_scope(from_datetime:, to_datetime:)
+            .where("#{sanitized_field_name} IS NOT NULL")
```

Comment:
Layout/MultilineMethodCallIndentation: Use 4 (not 2) spaces for indenting a condition in a `return` statement spanning multiple lines.